lib/fs.js

```diff
@@ -720,29 +720,47 @@ function fsyncSync(fd) {
   handleErrorFromBinding(ctx);
 }
 
-function mkdir(path, mode, callback) {
+function mkdir(path, options, callback) {
+  const defaults = { parent: false, mode: 0o777 };
```

Comment:
Do you think it could be beneficial to have this at the module level, instead of creating it within this function on every call? It may be optimized away, I'm not sure.

Reply:
For this, I like the destructuring with defaults pattern:
```js
const {
  parent = false,
  mode = 0o777
} = options;
```